src/client/src/components/register.js

```diff
@@ -0,0 +1,66 @@
+const React = require('react')
+const fetch = require('node-fetch')
+const {Link, Redirect} = require('react-router-dom')
+const responseCodes = require('../statusCodeResponses')
+
+const REGISTER_URL = '/register'
+
+class Register extends React.Component {
+    constructor(props){
+        super(props)
+
+        this.state = {
+            username: '',
+            password: '',
+            email: '',
+            successfulRegister: false
+        }
+        this.handleChange = this.handleChange.bind(this)
+        this.handleSubmit = this.handleSubmit.bind(this)
+    }
+    handleChange(event){
+        const settings = {'Username': 'username' , 'Password' : 'password', 'Email': 'email'}
+        const key = settings[event.target.placeholder]
+        this.setState({ [key] : event.target.value})
+    }
+    
+    handleSubmit(event){
+        fetch(REGISTER_URL, {
+            method: 'POST',
+            body: JSON.stringify(this.state),
+            headers: { 'Content-Type': 'application/json' }
+        })
+        .then(response => {
+            if(response.status === 200){
+                this.setState({successfulRegister: true})
+            }
+            if(responseCodes.hasOwnProperty(response.status)){
```

Comment:
Maybe just `_.isNumber(response.status)`. The property might be null or undefined as well.